src/main.c

```diff
@@ -412,11 +412,13 @@ static void set_shared_uss_env(ConfigManager *configmgr) {
 
     int length = index - thisEnv;
     char *key = malloc(length + 1);
+    memset(key, 0, sizeof(*key));
```

Comment:
`sizeof(*key)` is equivalent to `sizeof(char)` that is only 1 byte long. Please replace `sizeof(*key)` with `length + 1`.